disnake/embeds.py

```diff
@@ -302,6 +313,20 @@ def __bool__(self) -> bool:
             )
         )
 
+    def __eq__(self, other: Any) -> bool:
+        self_attributes_values, other_attributes_values = tuple(), tuple()  # noqa: C408
+        for slot in self.__slots__:
+            if slot == "_colour":
+                slot = "color"
+
+            self_attributes_values += (self.__getattribute__(slot) or None,)
+            other_attributes_values += (other.__getattribute__(slot) or None,)
+
+        return isinstance(other, Embed) and all(
```

Comment:
needs to do the isinstance check before everything else, otherwise we get this: 
![image](https://user-images.githubusercontent.com/71233171/191428868-df59ca54-c85e-4876-ba68-02258fc53926.png)

Reply:
At this point wouldn't it make sense to drop `all` and go with something like this, or am I missing something?
```py
if not isinstance(...): return

for slot in self.__slots__:
    if slot == "_colour": slot = "colour"
    if (getattr(self, slot) or None) == (getattr(other, slot) or None):
        return False
return True
```
(fwiw I haven't tested this)

Reply:
Agreed. Definitely better than looping once to "pre-process" the values, and then looping over it again via `all`.